### PR TITLE
[fix](ip) fix datatype serde for ipv6 with rowstore

### PR DESCRIPTION
**Pull request by amorynan**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
before this pr: 
if we has ipv6 column in table which is support store_row_column 
we insert some data
and then make sql with update we will meet core
```
*** Aborted at 1730367188 (unix time) try "date -d @1730367188" if you are using GNU date ***
*** Current BE git commitID: face753ded ***
*** SIGSEGV invalid permissions for mapped object (@0x60c000a339ae) received by PID 4176451 (TID 4187168 OR 0x7ff774f5a700) from PID 10697134; stack trace: ***
 0# doris::signal::(anonymous namespace)::FailureSignalHandler(int, siginfo_t*, void*) at /mnt/disk1/wangqiannan/amory/doris/be/src/common/signal_handler.h:421
 1# PosixSignals::chained_handler(int, siginfo*, void*) [clone .part.0] in /mnt/disk1/wangqiannan/tool/jdk-17.0.10/lib/server/libjvm.so
 2# JVM_handle_linux_signal in /mnt/disk1/wangqiannan/tool/jdk-17.0.10/lib/server/libjvm.so
 3# 0x00007FFFF63AEB50 in /lib64/libc.so.6
 4# doris::IPv6Value::from_string(unsigned __int128&, char const*, unsigned long) at /mnt/disk1/wangqiannan/amory/doris/be/src/vec/runtime/ipv6_value.h:55
 5# bool doris::vectorized::read_ipv6_text_impl<unsigned __int128>(unsigned __int128&, doris::vectorized::ReadBuffer&) at /mnt/disk1/wangqiannan/amory/doris/be/src/vec/io/io_helper.h:316
 6# doris::vectorized::DataTypeIPv6SerDe::read_one_cell_from_jsonb(doris::vectorized::IColumn&, doris::JsonbValue const*) const at /mnt/disk1/wangqiannan/amory/doris/be/src/vec/data_types/serde/data_type_ipv6_serde.cpp:76
 7# doris::vectorized::JsonbSerializeUtil::jsonb_to_block(std::vector<std::shared_ptr<doris::vectorized::DataTypeSerDe>, std::allocator<std::shared_ptr<doris::vectorized::DataTypeSerDe> > > const&, char const*, unsigned long, std::unordered_map<unsigned int, unsigned int, std::hash<unsigned int>, std::equal_to<unsigned int>, std::allocator<std::pair<unsigned int const, unsigned int> > > const&, doris::vectorized::Block&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::unordered_set<int, std::hash<int>, std::equal_to<int>, std::allocator<int> > const&) at /mnt/disk1/wangqiannan/amory/doris/be/src/vec/jsonb/serialize.cpp:104
 8# doris::vectorized::JsonbSerializeUtil::jsonb_to_block(std::vector<std::shared_ptr<doris::vectorized::DataTypeSerDe>, std::allocator<std::shared_ptr<doris::vectorized::DataTypeSerDe> > > const&, doris::vectorized::ColumnStr<unsigned int> const&, std::unordered_map<unsigned int, unsigned int, std::hash<unsigned int>, std::equal_to<unsigned int>, std::allocator<std::pair<unsigned int const, unsigned int> > > const&, doris::vectorized::Block&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, std::unordered_set<int, std::hash<int>, std::equal_to<int>, std::allocator<int> > const&) at /mnt/disk1/wangqiannan/amory/doris/be/src/vec/jsonb/serialize.cpp:83
 9# doris::BaseTablet::fetch_value_through_row_column(std::shared_ptr<doris::Rowset>, doris::TabletSchema const&, unsigned int, std::vector<unsigned int, std::allocator<unsigned int> > const&, std::vector<unsigned int, std::allocator<unsigned int> > const&, doris::vectorized::Block&) at /mnt/disk1/wangqiannan/amory/doris/be/src/olap/base_tablet.cpp:885
10# doris::FixedReadPlan::read_columns_by_plan(doris::TabletSchema const&, std::vector<unsigned int, std::allocator<unsigned int> >, std::map<doris::RowsetId, std::shared_ptr<doris::Rowset>, std::less<doris::RowsetId>, std::allocator<std::pair<doris::RowsetId const, std::shared_ptr<doris::Rowset> > > > const&, doris::vectorized::Block&, std::map<unsigned int, unsigned int, std::less<unsigned int>, std::allocator<std::pair<unsigned int const, unsigned int> > >*, signed char const*) const at /mnt/disk1/wangqiannan/amory/doris/be/src/olap/partial_update_info.cpp:295
11# doris::FixedReadPlan::fill_missing_columns(doris::RowsetWriterContext*, std::map<doris::RowsetId, std::shared_ptr<doris::Rowset>, std::less<doris::RowsetId>, std::allocator<std::pair<doris::RowsetId const, std::shared_ptr<doris::Rowset> > > > const&, doris::TabletSchema const&, doris::vectorized::Block&, std::vector<bool, std::allocator<bool> > const&, bool, unsigned long const&, doris::vectorized::Block const*) const at /mnt/disk1/wangqiannan/amory/doris/be/src/olap/partial_update_info.cpp:332
12# doris::segment_v2::VerticalSegmentWriter::_append_block_with_partial_content(doris::segment_v2::RowsInBlock&, doris::vectorized::Block&) at /mnt/disk1/wangqiannan/amory/doris/be/src/olap/rowset/segment_v2/vertical_segment_writer.cpp:502
```

### Check List (For Committer)

- Test 

    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No colde files have been changed.
        - [ ] Other reason 

- Behavior changed:

    - [x] No.
    - [ ] Yes. 

- Does this need documentation?

    - [x] No.
    - [ ] Yes. 

- Release note

    
    
    None

### Check List (For Reviewer who merge this PR)

- [x] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/test/vec/data_types/serde/data_type_serde_test.cpp [hunk 1]
```diff
@@ -220,4 +220,33 @@ TEST(DataTypeSerDeTest, DataTypeScalaSerDeTest) {
     serialize_and_deserialize_pb_test();
 }
```
Comment (eldenmoon) [inline be/test/vec/data_types/serde/data_type_serde_test.cpp:222]: also add groovy cases

Comment (amorynan) [reply be/test/vec/data_types/serde/data_type_serde_test.cpp:222]: done!

**Comment by amorynan** [comment]
run buildall

File: regression-test/suites/datatype_p0/ip/test_ip_basic.groovy [hunk 2]
```diff
@@ -146,4 +146,18 @@ suite("test_ip_basic") {
     sql "DROP TABLE t0"
     sql "DROP TABLE t1"
     sql "DROP TABLE t2"
+
+    // test ip with rowstore
+    sql """ SET enable_nereids_planner=true """
+    sql """ SET enable_fallback_to_original_planner=false """
+    sql """ DROP TABLE IF EXISTS table_ip """
+    sql """ CREATE TABLE IF NOT EXISTS `table_ip` (`col0` bigint NOT NULL,`col1` boolean NOT NULL, `col24` ipv4 NOT NULL, `col25` ipv6 NOT NULL,INDEX col1 (`col1`) USING INVERTED, INDEX col25 (`col25`) USING INVERTED ) ENGINE=OLAP UNIQUE KEY(`col0`) DISTRIBUTED BY HASH(`col0`) BUCKETS 4 PROPERTIES ("replication_allocation" = "tag.location.default: 1") """
```
Comment (eldenmoon) [inline regression-test/suites/datatype_p0/ip/test_ip_basic.groovy:154]: add store_row_column = true?

Comment (amorynan) [reply regression-test/suites/datatype_p0/ip/test_ip_basic.groovy]: done

**Comment by amorynan** [comment]
run buildall

**Review by xiaokang — DISMISSED** [review]
LGTM

File: be/src/vec/data_types/serde/data_type_ipv6_serde.cpp [hunk 3]
```diff
@@ -87,7 +87,10 @@ void DataTypeIPv6SerDe::write_one_cell_to_jsonb(const IColumn& column,
     result.writeKey(col_id);
     IPv6 data = assert_cast<const ColumnIPv6&>(column).get_element(row_num);
     IPv6Value ipv6_value(data);
-    result.writeString(ipv6_value.to_string());
+    std::string ipv6_str = ipv6_value.to_string();
```
Comment (yiguolei) [inline be/src/vec/data_types/serde/data_type_ipv6_serde.cpp:90]: it may cost too much
Could ipv6 value return StringRef directly?

**Review by eldenmoon — DISMISSED** [review]
LGTM

File: be/src/vec/data_types/serde/data_type_ipv6_serde.cpp (hunk 3 above)
Comment (eldenmoon) [reply be/src/vec/data_types/serde/data_type_ipv6_serde.cpp:90]: why not store int128 directly?

File: regression-test/suites/datatype_p0/ip/test_ip_basic.groovy [hunk 4]
```diff
@@ -146,4 +146,18 @@ suite("test_ip_basic") {
...
+    sql """ DROP TABLE IF EXISTS table_ip """
+    sql """ CREATE TABLE IF NOT EXISTS `table_ip` (`col0` bigint NOT NULL,`col1` boolean NOT NULL, `col24` ipv4 NOT NULL, `col25` ipv6 NOT NULL,INDEX col1 (`col1`) USING INVERTED, INDEX col25 (`col25`) USING INVERTED ) ENGINE=OLAP UNIQUE KEY(`col0`) DISTRIBUTED BY HASH(`col0`) BUCKETS 4 PROPERTIES ("replication_allocation" = "tag.location.default: 1", "store_row_column" = "true") """
+    sql """ insert into table_ip values (1, true, '255.255.255.255', "5be8:dde9:7f0b:d5a7:bd01:b3be:9c69:573b") """
+    qt_sql """ select * from table_ip """
+    sql """ Update table_ip set col1 = false where col0 = 1 """
+    qt_sql """ select * from table_ip """
+    sql """ Update table_ip set col24 = '127.0.0.1' where col0 = 1 """
+    qt_sql """ select * from table_ip """
+    sql """ Update table_ip set col25 = 'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff' where col0 = 1 """
+    qt_sql """ select * from table_ip """
```
Comment (eldenmoon) [inline regression-test/suites/datatype_p0/ip/test_ip_basic.groovy:162]: add case `select * from table_ip where col0 = 1`

Comment (eldenmoon) [reply regression-test/suites/datatype_p0/ip/test_ip_basic.groovy:162]: explain should contains `SHORT-CIRCUIT` flag

File: be/src/vec/data_types/serde/data_type_ipv6_serde.cpp (hunk 3 above)
Comment (amorynan) [reply be/src/vec/data_types/serde/data_type_ipv6_serde.cpp:90]: now we can use int128 directly

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Review by eldenmoon — DISMISSED** [review]
LGTM

File: be/src/vec/data_types/serde/data_type_ipv6_serde.cpp [hunk 5]
```diff
@@ -69,25 +69,17 @@ Status DataTypeIPv6SerDe::write_column_to_mysql(const IColumn& column,
...
 
 void DataTypeIPv6SerDe::write_one_cell_to_jsonb(const IColumn& column,
                                                 JsonbWriterT<JsonbOutStream>& result,
                                                 Arena* mem_pool, int col_id, int row_num) const {
-    // we make ipv6 as string in jsonb
     result.writeKey(col_id);
     IPv6 data = assert_cast<const ColumnIPv6&>(column).get_element(row_num);
-    IPv6Value ipv6_value(data);
-    result.writeString(ipv6_value.to_string());
+    result.writeInt128(int128_t(data));
```
Comment (yiguolei) [inline be/src/vec/data_types/serde/data_type_ipv6_serde.cpp:82]: using IPv6 = uint128_t;
If it is an uint128, why it could be convert to int128_t??

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — DISMISSED** [review]
LGTM

**Comment by amorynan** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run p0

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**